numpy/f2py/__init__.py

```diff
@@ -59,10 +59,29 @@ def compile(source,
         f.write(source)
         f.flush()
 
-        args = ' -c -m {} {} {}'.format(modulename, f.name, extra_args)
-        c = '{} -c "import numpy.f2py as f2py2e;f2py2e.main()" {}'
-        c = c.format(sys.executable, args)
-        status, output = exec_command(c)
+        # NOTE: extra_args is documented as a string,
+        # so we may want to split it, but tricky;
+        # standard library shlex.split() may be an interesting
+        # option if desired
+
+        args = ['-c', '-m', modulename, f.name, extra_args]
+
+        # we implicitly escape the quotes around the Python code
+        # in the following by nesting double inside single
+        c = [sys.executable,
+             '-c',
+             '"import numpy.f2py as f2py2e;f2py2e.main()"'] + args
```

Comment:
You need to remove these inner quotes. This is running a python program consisting of a single string literal.